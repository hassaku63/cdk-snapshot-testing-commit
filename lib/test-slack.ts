import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as aws_lambda from "aws-cdk-lib/aws-lambda";
import * as aws_codebuild from "aws-cdk-lib/aws-codebuild";

export class TestStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const func = new aws_lambda.Function(this, "TestFunction", {
      runtime: aws_lambda.Runtime.NODEJS_18_X,
      handler: "index.handler",
      code: aws_lambda.Code.fromInline(`
        exports.handler = async function (event, context) {
          console.log("EVENT: \n" + JSON.stringify(event, null, 2));
          return context.logStreamName;
        };
      `),
      environment: {
        "SLACK_CHANNEL": "test-channel",
      },
    });

    const project = new aws_codebuild.Project(this, "TestProject", {
      buildSpec: aws_codebuild.BuildSpec.fromObject({
        version: "0.2",
        phases: {
          build: {
            commands: [
              "echo hello world",
            ],
          },
        },
      }),
      environmentVariables: {
        "SLACK_CHANNEL": {
          type: aws_codebuild.BuildEnvironmentVariableType.PLAINTEXT,
          value: "test-channel",
        },
      },
    });

    new cdk.CfnOutput(this, "TestFunctionArn", { value: func.functionArn });
    new cdk.CfnOutput(this, "TestProjectArn", { value: project.projectArn });
  }
}
