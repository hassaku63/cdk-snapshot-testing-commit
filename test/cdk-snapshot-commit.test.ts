import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { TestStack } from '../lib/test-slack';

describe('Snapshot testing', () => {
  test('Test Stack', () => {
    const app = new cdk.App();
    const stack = new TestStack(app, 'MyTestStack', {
      // dummy value
      slackChannelId: '***',
    });

    const t = Template.fromStack(stack);
    expect(t).toMatchSnapshot();
  });
});
