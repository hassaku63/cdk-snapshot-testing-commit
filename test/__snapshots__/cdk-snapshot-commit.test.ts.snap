// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Test Stack 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "TestFunctionWithEnvADC2417D": {
  "Type": "AWS::Lambda::Function",
  "Properties": {
    "Code": {
      "ZipFile": "\\n        exports.handler = async function (event, context) {\\n          console.log(\\"EVENT: \\n\\" + JSON.stringify(event, null, 2));\\n          return context.logStreamName;\\n        };\\n      "
    },
    "Environment": {
      "Variables": {
        "SLACK_CHANNEL": "***"
      }
    },
    "Handler": "index.handler",
    "Role": {
      "Fn::GetAtt": [
        "TestFunctionWithEnvServiceRoleA6AA9548",
        "Arn"
      ]
    },
    "Runtime": "nodejs18.x"
  },
  "DependsOn": [
    "TestFunctionWithEnvServiceRoleA6AA9548"
  ]
},
    "TestFunctionWithEnvServiceRoleA6AA9548": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TestFunctionWithoutEnvF040F9FB": {
  "Type": "AWS::Lambda::Function",
  "Properties": {
    "Code": {
      "ZipFile": "\\n        exports.handler = async function (event, context) {\\n          console.log(\\"EVENT: \\n\\" + JSON.stringify(event, null, 2));\\n          return context.logStreamName;\\n        };\\n      "
    },
    "Handler": "index.handler",
    "Role": {
      "Fn::GetAtt": [
        "TestFunctionWithoutEnvServiceRole5D1C6DCC",
        "Arn"
      ]
    },
    "Runtime": "nodejs18.x"
  },
  "DependsOn": [
    "TestFunctionWithoutEnvServiceRole5D1C6DCC"
  ]
},
    "TestFunctionWithoutEnvServiceRole5D1C6DCC": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TestProjectWithEnv44BBE9D1": {
  "Type": "AWS::CodeBuild::Project",
  "Properties": {
    "Artifacts": {
      "Type": "NO_ARTIFACTS"
    },
    "Cache": {
      "Type": "NO_CACHE"
    },
    "EncryptionKey": "alias/aws/s3",
    "Environment": {
      "ComputeType": "BUILD_GENERAL1_SMALL",
      "EnvironmentVariables": [
        {
          "Name": "SLACK_CHANNEL",
          "Type": "PLAINTEXT",
          "Value": "***"
        }
      ],
      "Image": "aws/codebuild/standard:1.0",
      "ImagePullCredentialsType": "CODEBUILD",
      "PrivilegedMode": false,
      "Type": "LINUX_CONTAINER"
    },
    "ServiceRole": {
      "Fn::GetAtt": [
        "TestProjectWithEnvRoleA3D98701",
        "Arn"
      ]
    },
    "Source": {
      "BuildSpec": "{\\n  \\"version\\": \\"0.2\\",\\n  \\"phases\\": {\\n    \\"build\\": {\\n      \\"commands\\": [\\n        \\"echo hello world\\"\\n      ]\\n    }\\n  }\\n}",
      "Type": "NO_SOURCE"
    }
  }
},
    "TestProjectWithEnvRoleA3D98701": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TestProjectWithEnvRoleDefaultPolicy3EB0D709": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "TestProjectWithEnv44BBE9D1",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "TestProjectWithEnv44BBE9D1",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "TestProjectWithEnv44BBE9D1",
                    },
                    "-*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TestProjectWithEnvRoleDefaultPolicy3EB0D709",
        "Roles": [
          {
            "Ref": "TestProjectWithEnvRoleA3D98701",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TestProjectWithoutEnv5547A82C": {
  "Type": "AWS::CodeBuild::Project",
  "Properties": {
    "Artifacts": {
      "Type": "NO_ARTIFACTS"
    },
    "Cache": {
      "Type": "NO_CACHE"
    },
    "EncryptionKey": "alias/aws/s3",
    "Environment": {
      "ComputeType": "BUILD_GENERAL1_SMALL",
      "Image": "aws/codebuild/standard:1.0",
      "ImagePullCredentialsType": "CODEBUILD",
      "PrivilegedMode": false,
      "Type": "LINUX_CONTAINER"
    },
    "ServiceRole": {
      "Fn::GetAtt": [
        "TestProjectWithoutEnvRole84EA8F64",
        "Arn"
      ]
    },
    "Source": {
      "BuildSpec": "{\\n  \\"version\\": \\"0.2\\",\\n  \\"phases\\": {\\n    \\"build\\": {\\n      \\"commands\\": [\\n        \\"echo hello world\\"\\n      ]\\n    }\\n  }\\n}",
      "Type": "NO_SOURCE"
    }
  }
},
    "TestProjectWithoutEnvRole84EA8F64": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TestProjectWithoutEnvRoleDefaultPolicy6838AF7A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "TestProjectWithoutEnv5547A82C",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "TestProjectWithoutEnv5547A82C",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "TestProjectWithoutEnv5547A82C",
                    },
                    "-*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TestProjectWithoutEnvRoleDefaultPolicy6838AF7A",
        "Roles": [
          {
            "Ref": "TestProjectWithoutEnvRole84EA8F64",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
